daemon/start.go

```diff
@@ -177,15 +177,17 @@ func (daemon *Daemon) containerStart(container *container.Container, checkpoint
 		return err
 	}
 
-	ctx := context.TODO()
-
 	ctr, err := libcontainerd.ReplaceContainer(ctx, daemon.containerd, container.ID, spec, shim, createOptions)
 	if err != nil {
 		return translateContainerdStartErr(container.Path, container.SetExitCode, err)
 	}
 
+	// Passing ctx to ctr.Start caused integration tests
+	// to be stuck in the cleanup phase
+	ctrCtx := context.TODO()
+
 	// TODO(mlaventure): we need to specify checkpoint options here
-	tsk, err := ctr.Start(ctx, checkpointDir,
+	tsk, err := ctr.Start(ctrCtx, checkpointDir,
```

Comment:
```suggestion
	tsk, err := ctr.Start(context.TODO() /* Passing ctx caused integration tests to be stuck in the cleanup phase */, checkpointDir,
```
Please attach the TODO to the relevant line. It saves readers from having to chase variables around and check that they aren't used in unintended places.